spanner/api/Spanner/CancelBackupOperation.cs

```diff
@@ -0,0 +1,67 @@
+// Copyright 2020 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START spanner_cancel_backup_create]
+
+using Google.Cloud.Spanner.Admin.Database.V1;
+using Google.Cloud.Spanner.Common.V1;
+using Google.LongRunning;
+using Google.Protobuf.WellKnownTypes;
+using System;
+
+namespace GoogleCloudSamples.Spanner
+{
+    public class CancelBackupOperation
+    {
+        public static object SpannerCancelBackupOperation(
+            string projectId, string instanceId, string databaseId, string backupId)
+        {
+            // Create the DatabaseAdminClient instance.
+            DatabaseAdminClient databaseAdminClient = DatabaseAdminClient.Create();
+
+            // Initialize request parameters.
+            Backup backup = new Backup
+            {
+                Database = DatabaseName.Format(projectId, instanceId, databaseId),
+                ExpireTime = DateTime.UtcNow.AddDays(14).ToTimestamp()
+            };
+            InstanceName parentAsInstanceName = InstanceName.FromProjectInstance(projectId, instanceId);
+
+            // Make the CreateBackup request.
+            Operation<Backup, CreateBackupMetadata> operation =
+                databaseAdminClient.CreateBackup(parentAsInstanceName, backup, backupId);
+
+            // Cancel the operation.
+            operation.Cancel();
```

Comment:
What's the difference between canceling a backup and deleting one?
The docs for deleting say "Deletes a pending or completed backup", so I wonder if maybe canceling here is the same as deleting, and it would be best to just delete since we might need to do so anyway if the cancel didn't came in time?
Just asking, I'm fime with the sample as is.

Reply:
I guess the only difference is that delete is guaranteed to delete your backup while cancel is more of a best-effort operation and only applies if the backup is in the process of being created. If it's already created then it's a no-op.

Since the backup creation is a LRO, the backend team asked us to show how to cancel that LRO with this sample so for consistency with what's in the Cloud docs, we should leave this here as is.